src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1191,13 +1191,246 @@ class C
             }
             """;
         var comp = CreateCompilation(source, options: TestOptions.UnsafeDebugDll);
-        comp.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_RequiresLocationAttribute);
         comp.VerifyDiagnostics(
             // (3,36): error CS0518: Predefined type 'System.Runtime.CompilerServices.RequiresLocationAttribute' is not defined or imported
             //     public unsafe void M(delegate*<ref readonly int, void> p) { }
             Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "ref readonly int").WithArguments("System.Runtime.CompilerServices.RequiresLocationAttribute").WithLocation(3, 36));
     }
 
+    [Fact]
+    public void FunctionPointer_NoAttribute_PlusNormalMethod()
+    {
+        var source = """
+            class C
+            {
+                public unsafe void M1(delegate*<ref readonly int, void> p) { }
+                public void M2(ref readonly int p) { }
+            }
+            """;
+        var comp = CreateCompilation(source, options: TestOptions.UnsafeDebugDll);
```

Comment:
The core `CreateCompilation` helper is expected to roll forward to the latest target framework as we move forward. At the moment this references the `net7.0` assemblies but will move to `net8.0` (likely early 2024). At that point the test will fail because this will succeed since `RequiresLocationAttribute` will be available. Think this test should specify `TargetFramework.Net70` explicitly to avoid that future issue.

Reply:
Is that right? Seems that the default target framework is `netstandard2.0`. Otherwise, I would probably need to change a lot of tests in this file because for example all function pointer tests include the attribute definition as source.

Reply:
My mistake. Your correct this defaults to `TargteFramework.Standard` not `TargetFramework.Latest`.